desk/app/activity.hoon

```diff
@@ -770,50 +596,85 @@
   ^-  index:a
   =/  new-floor=(unit time)  (find-floor index)
   ?~  new-floor  index
-  index(floor.reads u.new-floor)
+  =/  new-reads=read-items:a
+    (lot:on-read-items:a items.reads.index new-floor ~)
+  index(reads [u.new-floor new-reads])
 ::
 ++  read
-  |=  [=source:a action=read-action:a]
+  |=  [=source:a action=read-action:a from-parent=?]
   ^+  cor
-  %+  update-reads  source
+  =/  =index:a  (get-index source)
   ?-  -.action
       %event
-    |=  =index:a
     ?>  ?=(%event -.action)
     =/  events
       %+  murn
         (tap:on-event:a stream.index)
       |=  [=time =event:a]
       ?.  =(-.event event.action)  ~
       `[time event]
-    ?~  events  index
-    =-  index(items.reads -)
-    %+  put:on-read-items:a  items.reads.index
-    [-<.events ~]
+    ?~  events  cor
+    (read source [%item -<.events] |)
   ::
       %item
-    |=  =index:a
-    =-  index(items.reads -)
-    %+  put:on-read-items:a  items.reads.index
-    [id.action ~]
+    =/  new-read  [id.action ~]
+    =/  read-items  (put:on-read-items:a items.reads.index new-read)
+    =.  cor  (update-parents source ~[new-read])
+    (update-index source index(items.reads read-items) &)
```

Comment:
Here and in `+read`, it looks like we're processing parents first, and then the index/source itself. But I thought we want to do children first, and then the parents? Or is that only relevant during summary calculation? iow: any reason we don't always go local->up?